### PR TITLE
[build-utils] add pnpm v8 auto detection

### PR DESCRIPTION
**Pull request by Ethan-Arrowood**
Adds automatic detection for pnpm v8 based on lockfile v6

### CODE REVIEW

File: packages/build-utils/src/fs/run-user-scripts.ts [hunk 1]
```diff
@@ -509,7 +513,18 @@ export function getEnvForPackageManager({
     ) {
       // Ensure that pnpm 7 is at the beginning of the `$PATH`
       newEnv.PATH = `${pnpm7}${path.delimiter}${oldPath}`;
-      console.log('Detected `pnpm-lock.yaml` generated by pnpm 7...');
+      console.log(
+        'Detected `pnpm-lock.yaml` version 5.4 generated by pnpm 7...'
```
Comment (TooTallNate) [inline packages/build-utils/src/fs/run-user-scripts.ts]: Should we use `lockfileVersion` in the string here?

File: packages/build-utils/test/fixtures/27-pnpm-v7/pnpm-workspace.yaml [hunk 2]
```diff
@@ -0,0 +1 @@
+# Empty so this is treated as its own pnpm module
```
Comment (TooTallNate) [inline packages/build-utils/test/fixtures/27-pnpm-v7/pnpm-workspace.yaml]: s/module/workspace?

File: packages/build-utils/src/fs/run-user-scripts.ts [hunk 3]
```diff
@@ -273,7 +273,10 @@ export async function scanParentDirs(
   } else if (pnpmLockYaml) {
     cliType = 'pnpm';
     lockfilePath = pnpmLockPath;
-    lockfileVersion = Number(pnpmLockYaml.lockfileVersion);
+    lockfileVersion =
+      typeof pnpmLockYaml.lockfileVersion === 'number'
+        ? pnpmLockYaml.lockfileVersion
+        : parseFloat(pnpmLockYaml.lockfileVersion);
```
Comment (styfle) [inline packages/build-utils/src/fs/run-user-scripts.ts]: Lets keep this `Number()`.

Its a little more strict compared to `parseFloat()`

For example, try with `"1a"`

**Review by styfle — DISMISSED** [review]
Thanks!

**Review by cb1kenobi — DISMISSED** [review]
Awesome!

File: packages/build-utils/test/fixtures/26-multiple-lock-files-pnpm/vercel.json [hunk 4]
```diff
@@ -4,7 +4,7 @@
   "probes": [
     {
       "path": "/",
-      "mustContain": "pnpm version: 6",
+      "mustContain": "pnpm version: 7",
       "logMustContain": "pnpm run build"
```
Comment (styfle) [inline packages/build-utils/test/fixtures/26-multiple-lock-files-pnpm/vercel.json]:
```suggestion
      "mustContain": "pnpm version: 7"
```
Lets simplify this

File: packages/build-utils/test/fixtures/27-pnpm-v7/probes.json [hunk 5]
```diff
@@ -0,0 +1,9 @@
+{
+  "probes": [
+    {
+      "path": "/",
+      "mustContain": "pnpm version: 7",
+      "logMustContain": "pnpm run build"
```
Comment (styfle) [inline packages/build-utils/test/fixtures/27-pnpm-v7/probes.json]:
```suggestion
      "mustContain": "pnpm version: 7"
```

Lets simplify this

File: packages/build-utils/test/fixtures/28-pnpm-v8/probes.json [hunk 6]
```diff
@@ -0,0 +1,9 @@
+{
+  "probes": [
+    {
+      "path": "/",
+      "mustContain": "pnpm version: 8",
+      "logMustContain": "pnpm run build"
```
Comment (styfle) [inline packages/build-utils/test/fixtures/28-pnpm-v8/probes.json]:
```suggestion
      "mustContain": "pnpm version: 8"
```

**Review by styfle — APPROVED** [review]

**Review by cb1kenobi — APPROVED** [review]

**Pull request merged**